### PR TITLE
Issue 4412: Fix bug in ExtendedS3StorageConfig to correctly handle missing trailing `/` in root.

### LINKED ISSUE #4412: ExtendedS3StorageConfig does not handle missing trailing `/` in root
**Problem description**
Current logic in ExtendedS3StorageConfig incorrectly assumes that root has a trailing '/' character. If this is not the case then the S3 storage is wrongly initialized and results in invalid paths/urls.

**Problem location**
ExtendedS3StorageConfig.java constructor where we initialize values from TypedProperties.

**Suggestions for an improvement**
Correctly handle the case, by adding the / char when it is missing.

### PR DESCRIPTION
**Pull request by sachin-j-joshi**
Current logic in ExtendedS3StorageConfig incorrectly assumes that root has a trailing '/' character. If this is not the case then the S3 storage is wrongly initialized and results in invalid paths/urls.
This change fixes the bug by adding the / char when it is missing.

Signed-off-by: Sachin Joshi <sachin.joshi@emc.com>

**Change log description**  
Fix bug in ExtendedS3StorageConfig to correctly handle missing trailing `/` in root.

**Purpose of the change**  
Fixes #4412 

**What the code does**  
This change fixes it by adding the / char when it is missing.

**How to verify it**  
All tests should pass.

### CODE REVIEW

File: bindings/src/test/java/io/pravega/storage/extendeds3/ExtendedS3StorageTest.java [hunk 1]
```diff
@@ -80,6 +84,23 @@ public void testCreateIfNoneMatch() {
     }
     //endregion
 
+    @Test
```
Comment (kevinhan88) [inline bindings/src/test/java/io/pravega/storage/extendeds3/ExtendedS3StorageTest.java]: Suggest to add case to ensure the tailing space could also be trimmed.

File: bindings/src/main/java/io/pravega/storage/extendeds3/ExtendedS3StorageConfig.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
     public static final Property<Boolean> USENONEMATCH = Property.named("useNoneMatch", false);
 
     private static final String COMPONENT_CODE = "extendeds3";
+    private static final String PATH_SEPARATOR = "/";
```
Comment (andreipaduroiu) [inline bindings/src/main/java/io/pravega/storage/extendeds3/ExtendedS3StorageConfig.java]: This should go into `StreamSegmentNameUtils.java` and you should update all references to the string `"/"` in there too.

Comment (sachin-j-joshi) [reply bindings/src/main/java/io/pravega/storage/extendeds3/ExtendedS3StorageConfig.java]: StreamSegmentNameUtils is in ':shared:protocol' project which has mostly protocol related stuff. Today 'bindings' project does not depend on it and it doesn't make sense to bring in that dependency just for this change.

File: bindings/src/test/java/io/pravega/storage/extendeds3/ExtendedS3StorageTest.java (hunk 1 above)
Comment (sachin-j-joshi) [reply bindings/src/test/java/io/pravega/storage/extendeds3/ExtendedS3StorageTest.java]: Added test

**Pull request merged**